lib/api/pokedex_api_spec/handler/pokemon_handler.dart

```diff
@@ -18,7 +18,13 @@ class PokemonApi {
 
     final baseUri = Uri.parse(apiClient.dio.options.baseUrl);
     final uri = baseUri.replace(queryParameters: queryParams, path: '${baseUri.path}/pokemon');
-    return await apiClient.dio.getUri(uri).then(
-        (response) => response.data[str.results].map<Pokemon>((dynamic e) => Pokemon.fromJson(e as Json)).toList());
+    return await apiClient.dio.getUri(uri).then((response) => response.data[str.results]
+        .map<Pokemon>(
```

Comment:
why do manual mapping here instead of utilizing your `fromJson` factory method? 🤔 

perhaps you can still improve this like this one:

```
return await apiClient.dio.getUri(uri).then((response) {
      return response.data[str.results].map<Pokemon>((dynamic data) => Pokemon.fromJson(data as Json)).toList();
    });
```